jenkins/spark-premerge-build.sh

```diff
@@ -206,7 +206,7 @@ ci_scala213() {
     cd .. # Run integration tests in the project root dir to leverage test cases and resource files
     export TEST_TAGS="not premerge_ci_1"
     export TEST_TYPE="pre-commit"
-    export TEST_PARALLEL=5
+    export TEST_PARALLEL=4
```

Comment:
This doesn't seem like a change that should be made as part of Delta Lake low shuffle merge but rather as a separate PR, especially if you can get it to fail without your low shuffle merge changes.  cc: @NvTimLiu for visibility.

Reply:
What I'm saying above is that this change is not really related to this PR.  It's a significant change in CI scripts that will affect performance of CI, since we'll run fewer tests in parallel.  That's why I think this should be a separate change, not hidden in a large PR as a side-effect.